internal/seed/seed.go

```diff
@@ -0,0 +1,194 @@
+		fmt.Sprintf("API_EXTERNAL_URL=http://%s:%d", utils.Config.Hostname, utils.Config.Api.Port),
+
+		"GOTRUE_API_HOST=0.0.0.0",
+		"GOTRUE_API_PORT=9999",
+		"GOTRUE_TRACING_ENABLED=false",
+		"GOTRUE_METRICS_ENABLED=false",
+
+		"GOTRUE_DB_DRIVER=postgres",
+		"GOTRUE_DB_NAMESPACE=postgres",
+		fmt.Sprintf("GOTRUE_DB_DATABASE_URL=%s", source),
+
+		"GOTRUE_SITE_URL=" + utils.Config.Auth.SiteUrl,
+
+		"GOTRUE_JWT_ADMIN_ROLES=service_role",
+		"GOTRUE_JWT_AUD=authenticated",
+		"GOTRUE_JWT_DEFAULT_GROUP_NAME=authenticated",
+		fmt.Sprintf("GOTRUE_JWT_EXP=%v", utils.Config.Auth.JwtExpiry),
+		"GOTRUE_JWT_SECRET=" + utils.Config.Auth.JwtSecret,
+		fmt.Sprintf("GOTRUE_JWT_ISSUER=http://%s:%d/auth/v1", utils.Config.Hostname, utils.Config.Api.Port),
+	}
+
+	gotrueContainerId, err := utils.DockerStart(
+		ctx,
+		container.Config{
+			Image:        utils.Config.Auth.Image,
+			Env:          env,
+			ExposedPorts: nat.PortSet{"9999/tcp": {}},
+			Healthcheck: &container.HealthConfig{
+				Test:     []string{"CMD", "wget", "--no-verbose", "--tries=1", "--spider", "http://127.0.0.1:9999/health"},
+				Interval: 10 * time.Second,
+				Timeout:  2 * time.Second,
+				Retries:  3,
+			},
+		},
+		container.HostConfig{
+			PortBindings:  nat.PortMap{"9999/tcp": []nat.PortBinding{{HostPort: "54325"}}},
+			RestartPolicy: container.RestartPolicy{Name: "always"},
+		},
+		network.NetworkingConfig{
+			EndpointsConfig: map[string]*network.EndpointSettings{
+				utils.NetId: {
+					Aliases: utils.GotrueAliases,
+				},
+			},
+		},
+		fmt.Sprintf("%s_%s", utils.GotrueId, "shadow"),
+	)
+	if err != nil {
+		return err
+	}
+	defer utils.DockerRemove(gotrueContainerId)
+	if err := reset.WaitForServiceReady(ctx, []string{gotrueContainerId}); err != nil {
+		return err
+	}
+
+	// call admin endpoint to create some test users
+	if err := createTestUsers(ctx, numberOfTestUsersCreated); err != nil {
+		return err
+	}
+
+	// write dump out to supabase/seed.example.sql file
+	outStream, err := fsys.OpenFile(utils.SeedExampleDataPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return errors.Errorf("failed to open seed example file: %w", err)
+	}
+	defer outStream.Close()
+
+	// dump data from shadow db
+	if err := dump.DumpData(ctx, pgconn.Config{
+		Host:     utils.Config.Hostname,
+		Port:     uint16(utils.Config.Db.ShadowPort),
+		User:     "postgres",
+		Password: utils.Config.Db.Password,
+		Database: "postgres",
+	}, []string{"auth"}, []string{
+		// auth schema needs to be specified to exclude the audit_log_entries table
+		"auth.audit_log_entries",
+		"instances",
+		"sessions",
+		"schema_migrations",
+		"refresh_tokens",
+		"mfa_amr_claims",
+		"mfa_challenges",
+		"mfa_factors",
+		"saml_providers",
+		"sso_providers",
+		"sso_domains",
+		"saml_relay_state",
+		"flow_state",
+	}, false, false, outStream); err != nil {
+		return err
+	}
+
+	// use awk to remove unnecessary sections from the dump
+	awkCmd := exec.Command("awk", `
```

Comment:
We can't expect `awk` to always be installed on user's system, ie. windows?

We can either process the docker output in go or make a shell script that runs inside gotrue container.